File: pbweberog/pbweberog-jar/src/main/java/it/csi/pbandi/pbweberog/business/service/FideiussioniService.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbweberog.business.service;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.Consumes;
import javax.ws.rs.DELETE;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.QueryParam;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import it.csi.pbandi.pbservizit.exception.UtenteException;
import it.csi.pbandi.pbweberog.dto.CodiceDescrizione;
import it.csi.pbandi.pbweberog.dto.EsitoOperazioni;
import it.csi.pbandi.pbweberog.dto.fideiussioni.Fideiussione;
import it.csi.pbandi.pbweberog.dto.fideiussioni.FideiussioneDTO;
import it.csi.pbandi.pbweberog.dto.fideiussioni.FideiussioneEsitoGenericoDTO;
import it.csi.pbandi.pbweberog.integration.request.CercaFideiussioniRequest;
import it.csi.pbandi.pbweberog.integration.request.CreaAggiornaFideiussioneRequest;

@Path("/fideiussioni")
@Api(value = "fideiussioni")
public interface FideiussioniService {


	@GET
	@Path("/home") 
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Verify first that the Fideiussione is manageable", response=EsitoOperazioni.class)
	Response isFideiussioneGestibile( @Context HttpServletRequest req, @QueryParam("idProgetto") Long idProgetto) throws UtenteException, Exception;
	
	@GET
	@Path("/tipiDiTrattamento") 
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Finds Tipi Di Trattamento (load combo tipo trattamento) ", response=CodiceDescrizione.class, responseContainer = "List")
	Response getTipiDiTrattamento( @Context HttpServletRequest req, @QueryParam("idProgetto") Long idProgetto) throws UtenteException, Exception;
	
	@GET
	@Path("/tipiDiTrattamento/{idTipoTrattamento}") 
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Finds Tipo Di Trattamento by idTipoTrattamento(for dettaglio Fideiussione) ", response=String.class)
	Response getTipoDiTrattamento( @Context HttpServletRequest req, @PathParam("idTipoTrattamento") Long idTipoTrattamento) throws UtenteException, Exception;
	
	
	@POST
	@Path("/") 
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Finds Fideiussioni by idProgetto and FiltroRicercaFideiussione", response=Fideiussione.class, responseContainer = "List")
	Response getFideiussioni( @Context HttpServletRequest req, CercaFideiussioniRequest cercaRequest) throws UtenteException, Exception;
	
	
	@DELETE
	@Path("/{idFideiussione}") 
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Deletes Fideiussione by idFideiussione", response=FideiussioneEsitoGenericoDTO.class)
	Response eliminaFideiussione( @Context HttpServletRequest req, @PathParam("idFideiussione") Long idFideiussione) throws UtenteException, Exception;
	
	

	@PUT
	@Path("/") 
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Updates Fideiussione", response=FideiussioneEsitoGenericoDTO.class)
	Response creaAggiornaFideiussione( @Context HttpServletRequest req, CreaAggiornaFideiussioneRequest moRequest) throws UtenteException, Exception;
	
	@POST
	@Path("/{idFideiussione}") 
	@Produces(MediaType.APPLICATION_JSON)
	@ApiOperation(value = "Finds dettaglio fideiussione", response=Fideiussione.class)
	Response dettaglioFideiussione( @Context HttpServletRequest req, @PathParam("idFideiussione") Long idFideiussione) throws UtenteException, Exception;
	
	
	
}

File: pbweberogwcl/src/app/fideiussioni/services/fideiussioni.service.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

/* tslint:disable */
import { Injectable } from '@angular/core';
import { HttpClient, HttpRequest, HttpResponse, HttpHeaders } from '@angular/common/http';
import { BaseService as __BaseService } from '../../shared/base-service';
// @ts-ignore
import { ApiConfiguration as __Configuration } from '../../shared/api-configuration';
import { StrictHttpResponse as __StrictHttpResponse } from '../../shared/strict-http-response'
import { Observable as __Observable } from 'rxjs';
import { map as __map, filter as __filter } from 'rxjs/operators';
import {CercaFideiussioniRequest} from "../commons/dto/cerca-fideiussioni-request";
import {Fideiussione} from "../commons/dto/fideiussione";
import {CreaAggiornaFideiussioneRequest} from "../commons/dto/crea-aggiorna-fideiussione-request";
import {FideiussioneEsitoGenericoDTO} from "../commons/dto/fideiussione-esito-generico-dto";
import {ConfigService} from "../../core/services/config.service";
import {FideiussioneDTO} from "../commons/dto/fideiussione-dto";
import {EsitoOperazioni} from "../commons/dto/esito-operazioni";
import {CodiceDescrizione} from "../commons/dto/codice-descrizione";



@Injectable({
  providedIn: 'root',
})
class FideiussioniService extends __BaseService {
  static readonly getFideiussioniPath = '/fideiussioni';
  static readonly creaAggiornaFideiussionePath = '/fideiussioni';
  static readonly dettaglioFideiussionePath = '/fideiussioni/{idFideiussione}';
  static readonly eliminaFideiussionePath = '/fideiussioni/{idFideiussione}';

  constructor(
    config: __Configuration,
    configService: ConfigService,
    http: HttpClient
  ) {
    super(config, configService, http);
  }

  /**
   * Verify first that the Fideiussione is manageable
   * @param idProgetto undefined
   * @return successful operation
   */
  isFideiussioneGestibileResponse(idProgetto?: number): __Observable<__StrictHttpResponse<EsitoOperazioni>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    if (idProgetto != null) __params = __params.set('idProgetto', idProgetto.toString());
    let req = new HttpRequest<any>(
      'GET',
      this.rootUrl + `/restfacade/fideiussioni/home`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<EsitoOperazioni>;
      })
    );
  }
  /**
   * Verify first that the Fideiussione is manageable
   * @param idProgetto undefined
   * @return successful operation
   */
  isFideiussioneGestibile(idProgetto?: number): __Observable<EsitoOperazioni> {
    return this.isFideiussioneGestibileResponse(idProgetto).pipe(
      __map(_r => _r.body as EsitoOperazioni)
    );
  }

  /**
   * Finds Tipi Di Trattamento (load combo tipo trattamento)
   * @param idProgetto undefined
   * @return successful operation
   */
  getTipiDiTrattamentoResponse(idProgetto?: number): __Observable<__StrictHttpResponse<Array<CodiceDescrizione>>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    if (idProgetto != null) __params = __params.set('idProgetto', idProgetto.toString());
    let req = new HttpRequest<any>(
      'GET',
      this.rootUrl + `/restfacade/fideiussioni/tipiDiTrattamento`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<Array<CodiceDescrizione>>;
      })
    );
  }
  /**
   * Finds Tipi Di Trattamento (load combo tipo trattamento)
   * @param idProgetto undefined
   * @return successful operation
   */
  getTipiDiTrattamento(idProgetto?: number): __Observable<Array<CodiceDescrizione>> {
    return this.getTipiDiTrattamentoResponse(idProgetto).pipe(
      __map(_r => _r.body as Array<CodiceDescrizione>)
    );
  }

  /**
   * Finds Fideiussioni by idProgetto and FiltroRicercaFideiussione
   * @param body undefined
   * @return successful operation
   */
  getFideiussioniResponse(body?: CercaFideiussioniRequest): __Observable<__StrictHttpResponse<Array<Fideiussione>>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    __body = body;
    let req = new HttpRequest<any>(
      'POST',
      this.rootUrl + `/restfacade/fideiussioni`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<Array<Fideiussione>>;
      })
    );
  }
  /**
   * Finds Fideiussioni by idProgetto and FiltroRicercaFideiussione
   * @param body undefined
   * @return successful operation
   */
  getFideiussioni(body?: CercaFideiussioniRequest): __Observable<Array<Fideiussione>> {
    return this.getFideiussioniResponse(body).pipe(
      __map(_r => _r.body as Array<Fideiussione>)
    );
  }

  /**
   * Updates Fideiussione
   * @param body undefined
   * @return successful operation
   */
  creaAggiornaFideiussioneResponse(body?: CreaAggiornaFideiussioneRequest): __Observable<__StrictHttpResponse<FideiussioneEsitoGenericoDTO>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;
    __body = body;
    let req = new HttpRequest<any>(
      'PUT',
      this.rootUrl + `/restfacade/fideiussioni`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<FideiussioneEsitoGenericoDTO>;
      })
    );
  }
  /**
   * Updates Fideiussione
   * @param body undefined
   * @return successful operation
   */
  creaAggiornaFideiussione(body?: CreaAggiornaFideiussioneRequest): __Observable<FideiussioneEsitoGenericoDTO> {
    return this.creaAggiornaFideiussioneResponse(body).pipe(
      __map(_r => _r.body as FideiussioneEsitoGenericoDTO)
    );
  }

  /**
   * Finds dettaglio fideiussione
   * @param idFideiussione undefined
   * @return successful operation
   */
  dettaglioFideiussioneResponse(idFideiussione: number): __Observable<__StrictHttpResponse<Fideiussione>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;

    let req = new HttpRequest<any>(
      'POST',
      this.rootUrl + `/restfacade/fideiussioni/${encodeURIComponent(String(idFideiussione))}`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<Fideiussione>;
      })
    );
  }
  /**
   * Finds dettaglio fideiussione
   * @param idFideiussione undefined
   * @return successful operation
   */
  dettaglioFideiussione(idFideiussione: number): __Observable<Fideiussione> {
    return this.dettaglioFideiussioneResponse(idFideiussione).pipe(
      __map(_r => _r.body as Fideiussione)
    );
  }

  /**
   * Deletes Fideiussione by idFideiussione
   * @param idFideiussione undefined
   * @return successful operation
   */
  eliminaFideiussioneResponse(idFideiussione: number): __Observable<__StrictHttpResponse<FideiussioneEsitoGenericoDTO>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;

    let req = new HttpRequest<any>(
      'DELETE',
      this.rootUrl + `/restfacade/fideiussioni/${encodeURIComponent(String(idFideiussione))}`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'json'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<FideiussioneEsitoGenericoDTO>;
      })
    );
  }
  /**
   * Deletes Fideiussione by idFideiussione
   * @param idFideiussione undefined
   * @return successful operation
   */
  eliminaFideiussione(idFideiussione: number): __Observable<FideiussioneEsitoGenericoDTO> {
    return this.eliminaFideiussioneResponse(idFideiussione).pipe(
      __map(_r => _r.body as FideiussioneEsitoGenericoDTO)
    );
  }

  /**
   * Finds Tipo Di Trattamento by idTipoTrattamento(for dettaglio Fideiussione)
   * @param idTipoTrattamento undefined
   * @return successful operation
   */
  getTipoDiTrattamentoResponse(idTipoTrattamento: number): __Observable<__StrictHttpResponse<string>> {
    let __params = this.newParams();
    let __headers = new HttpHeaders();
    let __body: any = null;

    let req = new HttpRequest<any>(
      'GET',
      this.rootUrl + `/restfacade/fideiussioni/tipiDiTrattamento/${encodeURIComponent(String(idTipoTrattamento))}`,
      __body,
      {
        headers: __headers,
        params: __params,
        responseType: 'text'
      });

    return this.http.request<any>(req).pipe(
      __filter(_r => _r instanceof HttpResponse),
      __map((_r) => {
        return _r as __StrictHttpResponse<string>;
      })
    );
  }
  /**
   * Finds Tipo Di Trattamento by idTipoTrattamento(for dettaglio Fideiussione)
   * @param idTipoTrattamento undefined
   * @return successful operation
   */
  getTipoDiTrattamento(idTipoTrattamento: number): __Observable<string> {
    return this.getTipoDiTrattamentoResponse(idTipoTrattamento).pipe(
      __map(_r => _r.body as string)
    );
  }

}

module FideiussioniService {
}

export { FideiussioniService }
